packages/button/src/Button.js

```diff
@@ -26,6 +26,8 @@ type Props = {
   disabled?: boolean,
   /** Stretch Button to fill its container */
   fullWidth?: boolean,
+  /** An instance of an Icon component */
+  icon?: MnrlReactNode,
```

Comment:
According to the designs, Buttons need to support icons at the start and the end.  We should verify this.  Perhaps the design is illustrating some other sort of component, such as a SplitButton?